### PR TITLE
When showing measurements on multiple dates, also allow for having co…

### LINKED ISSUE #7039: Add delta column
# User story

As a quality manager I want to see the difference between measurements made at two or more dates.

## Acceptance criteria

- [ ] User can turn delta's on and off via the settings panel
- [ ] The delta setting is included in the URL parameter
- [ ] When showing multiple dates a delta column is added between each pair of dates, showing the delta between each pair of dates
- [ ] The delta column shows the measurement difference as:
  - [ ]  "+&lt;number&gt;" or "-&lt;number&gt;" in case of count and percentage scale (note that the percentage delta is not a percentage)
  - [ ]  "lower", "equal", "higher" in case of version numbers
- [ ] The delta column display the delta using a green label if the measurement is better and a red label if it's worse, no label if it's equal. 
- [ ] The delta column header is "Change"

## Nice to have

- [ ] When showing one date, a delta column is added, showing the delta with yesterday's latest measurements

## Assumptions

- No changes needed to the API.

## Open issues

- What do we need to ...

# Tasks

### Frontend

- [ ] Add columns
- [ ] Add setting

## Documentation

- [ ] Update changelog
- [ ] Add instruction to user manual
- [ ] Update screenshots

### PR DESCRIPTION
**Pull request by fniessink**
…lumns that show the delta between dates. The delta columns can be turned on and off via the Settings panel. Closes #7039.

### CODE REVIEW

File: components/frontend/src/sharedPropTypes.js [hunk 1]
```diff
@@ -110,6 +110,8 @@ export const measurementSourceType = PropTypes.shape({
     parse_error: PropTypes.string,
 })
 
+export const measurementValueType = PropTypes.oneOfType([PropTypes.number, PropTypes.string])
```
Comment (wkoot) [inline components/frontend/src/sharedPropTypes.js]: Why can it be number, doesn't the api always return strings? Or am I confusing this with collector measurement tests?

File: components/frontend/src/subject/SubjectTableHeader.js [hunk 2]
```diff
@@ -225,6 +225,31 @@ const tagsHelp = <>
...
+    const cells = []
+    columnDates.forEach((date, index) => {
+        if (showDeltaColumns && index > 0) {
+            cells.push(
+                <UnsortableTableHeaderCell
+                    key={`delta-${date}`}
+                    help="The difference between the measurement values on the previous and next date.
+                    A plus (+) sign indicates that the newer value is higher, a minus (-) sign that it is lower.
+                    A green outline indicates that the newer value is better, a red outline that it is worse."
+                    label="𝚫"
```
Comment (wkoot) [inline components/frontend/src/subject/SubjectTableHeader.js:238]: Is this the ASCII capital delta, or a unicode char?

Comment (wkoot) [reply components/frontend/src/subject/SubjectTableHeader.js:238]: The ASCII version should be a Δ, which shows different from 𝚫

File: components/frontend/src/subject/SubjectTableRow.js [hunk 3]
```diff
@@ -25,28 +26,97 @@ import {
...
+        const improved = didValueImprove(increased, direction)
+        const evaluateTarget = metric.evaluate_targets ?? true
+        let alt = "The measurement value changed";
+        let color = "blue";
+         if (evaluateTarget && improved) {
+            alt = "The measurement value improved";
+            color = "green"
+        }
+        if (evaluateTarget && !improved) {
+            alt = "The measurement value worsened";
```
Comment (wkoot) [inline components/frontend/src/subject/SubjectTableRow.js:70]: Is it okay to not address the case where the value stayed the same? Do we skip that?

File: components/frontend/src/subject/SubjectTableRow.js [hunk 4]
```diff
@@ -25,28 +26,97 @@ import {
...
+            color = "red"
+        }
+        let delta = increased ? "+" : "-"
```
Comment (wkoot) [inline components/frontend/src/subject/SubjectTableRow.js:73]: This one also omits the case of value staying the same

File: components/frontend/src/subject/SubjectTableRow.test.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+import { SubjectTableRow } from './SubjectTableRow';
+import { createTestableSettings, datamodel, report } from "../__fixtures__/fixtures";
+
+beforeEach(() => {
+    history.push("")
+})
+
+function renderSubjectTableRow(
+    {
+        direction = "<",
```
Comment (wkoot) [inline components/frontend/src/subject/SubjectTableRow.test.js:14]: Seems useful to also test the opposite direction

File: components/frontend/src/subject/SubjectTableRow.js [hunk 6]
```diff
@@ -25,28 +26,97 @@ import {
...
+        if (metric.scale !== "version_number") {
+            delta += `${Math.abs(metricValue - previousValue)}`
+            alt += ` with ${delta}`
```
Comment (wkoot) [inline components/frontend/src/subject/SubjectTableRow.js]:
```suggestion
            alt += ` by ${delta}`
```

Comment (wkoot) [reply components/frontend/src/subject/SubjectTableRow.js]: I think "worsened by", "improved by" and "changed by" are more accurate when dealing with numeric values

File: components/frontend/src/sharedPropTypes.js (hunk 1 above)
Comment (fniessink) [reply components/frontend/src/sharedPropTypes.js]: Good catch. I was assuming number values had already been converted, but I was hallucinating like a LLM.

File: components/frontend/src/subject/SubjectTableHeader.js (hunk 2 above)
Comment (fniessink) [reply components/frontend/src/subject/SubjectTableHeader.js:238]: It's a unicode char. Should work fine.

File: components/frontend/src/subject/SubjectTableRow.js (hunk 3 above)
Comment (fniessink) [reply components/frontend/src/subject/SubjectTableRow.js:70]: Yes, in that case the delta cell is empty.

File: components/frontend/src/subject/SubjectTableRow.js (hunk 4 above)
Comment (fniessink) [reply components/frontend/src/subject/SubjectTableRow.js:73]: As designed, I'll add a comment.

File: components/frontend/src/subject/SubjectTableRow.test.js (hunk 5 above)
Comment (fniessink) [reply components/frontend/src/subject/SubjectTableRow.test.js:14]: That is done, that's why it's a parameter here.

**Review by wkoot — APPROVED** [review]

**Pull request merged**